artifactory_test.go

```diff
@@ -202,6 +202,29 @@ func TestArtifactoryDownloadPathWithSpecialChars(t *testing.T) {
 	cleanArtifactoryTest()
 }
 
+func TestArtifactoryDownloadPatternWithUnicodeChars(t *testing.T) {
+	initArtifactoryTest(t)
+	artifactoryCli.Exec("upload", "testsdata/unicode/", tests.Repo1, "--flat=false")
+
+	// Verify files exist
+	specFile, err := tests.CreateSpec(tests.SearchAllRepo1)
+	assert.NoError(t, err)
+	verifyExistInArtifactory(tests.GetDownloadUnicode(), specFile, t)
+
+	artifactoryCli.Exec("dl", tests.Repo1+"/testsdata/unicode/(dirλrec၇tory)/", tests.Out+fileutils.GetFileSeparator()+"{1}"+fileutils.GetFileSeparator(), "--flat=true")
+
+	paths, _ := fileutils.ListFilesRecursiveWalkIntoDirSymlink(tests.Out, false)
+	err = tests.ValidateListsIdentical([]string{
+		tests.Out,
+		tests.Out + fileutils.GetFileSeparator() + "dirλrec၇tory",
+		tests.Out + fileutils.GetFileSeparator() + "dirλrec၇tory" + fileutils.GetFileSeparator() + "文件.in",
+		tests.Out + fileutils.GetFileSeparator() + "dirλrec၇tory" + fileutils.GetFileSeparator() + "aȩ.ȥ1",
```

Comment:
Let's use https://golang.org/pkg/path/filepath/#Join instead of fileutils.GetFileSeparator() in this function. This may be a good opportunity to make this replace in the entire file.

Reply:
Fixed.